fix: (platform) add switch component

Betrozov opened the pull request:
#### Please provide a link to the associated issue.
fixes https://github.com/SAP/fundamental-ngx/issues/2867

#### Please provide a brief summary of this pull request.
Created the Platform Switch component

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [X] Documentation Examples
- [x] Stackblitz works for all examples

fkolar commented on apps/docs/src/app/platform/component-docs/platform-forms/switch/switch-header/switch-header.component.html:
```diff
@@ -0,0 +1,8 @@
+<description>
+    <p>Switch description</p>
+</description>
+<import module="SwitchModule"></import>
```
Missing actual description what we usually add to API docs:

Checkout other components in the platform where we not try to say the same what you can read in core but at minimum how we are enhancing this.

Betrozov replied:
fixed

fkolar commented on libs/platform/src/lib/components/form/switch/switch/switch.config.ts:
```diff
@@ -0,0 +1,24 @@
+export class SwitchConfig {
+    /**
+     * Content Density of element. 'cozy' | 'compact'
+     */
```
We could have more generic config as content density can be applied to all the components like this. I remember Dima was showing me `PlatformConfig`

valorkin replied:
It wasn't yet approved, so let's stick with component level configs, and later we could came up with platform/core/ or even all level configs and services

fkolar replied:
sounds good if its one component but if we need to keep config for each new component like this, I would go in the next iteration to more generic approach and refactor this.

valorkin replied:
components specific configs are must have anyway, over time they will have even more properties

JKMarkowski commented:
Hi @Betrozov,
I see that whole switch component is implemented from scratch, wasn't it supposed to use implementation from core?

fkolar commented:
There is missing disabled case in fundamentals- styles

JKMarkowski commented on libs/platform/src/lib/components/form/switch/switch/switch.component.ts:
```diff
@@ -0,0 +1,129 @@
+export class SwitchComponent extends BaseInput {
+    /** aria-label attribute of the inner input element. */
+    @Input()
+    ariaLabel = 'Switch input';
```
I know it's detail, but could you add type there?

fkolar replied:
In this you dont really need to. as you assign a string, this is default behavior for Lint Rules. We dont have general guidelines for this and this is some thing to be noted as well and decide on the direction. I would leave this as it is as this is also proper form.

JKMarkowski commented on libs/platform/src/lib/components/form/switch/switch/switch.component.ts:
```diff
@@ -0,0 +1,129 @@
+
+    /** Whether the switch is semantic */
+    @Input()
+    semantic = false;
```
Let's add missing type

Betrozov commented on libs/platform/src/lib/components/form/switch/switch/index.ts:
```diff
@@ -0,0 +1 @@
+export * from './switch.component';
```
it is for decreasing the path when we import this component.

fkolar replied:
Its common practice to use barrel files.

Betrozov commented on apps/docs/src/app/platform/platform-documentation.routes.ts:
```diff
@@ -4,91 +4,97 @@ import { PlatformHomeComponent } from './component-docs/platform-home/platform-h
-            }
-        ]
-    }
+  {
```
fixed

Betrozov commented:
> The styling for disabled looks not proper
> in platform
> <img alt="Screen Shot 2020-07-28 at 7 33 32 AM" width="215" src="https://user-images.githubusercontent.com/4380815/88660411-aa623c00-d0a4-11ea-961c-1dc1ff4b7c19.png">
> 
> in core
> <img alt="Screen Shot 2020-07-28 at 7 31 57 AM" width="186" src="https://user-images.githubusercontent.com/4380815/88660386-9a4a5c80-d0a4-11ea-918c-037935205118.png">

fixed

Betrozov commented:
> I added some comments regarding code.
> There is also different focus ring on switch in platform.
> Core:
> ![image](https://user-images.githubusercontent.com/26483208/88663235-89690d80-d0db-11ea-86aa-8e51be0ae78f.png)
> 
> Platform:
> ![image](https://user-images.githubusercontent.com/26483208/88663273-9d147400-d0db-11ea-916f-328d4639458c.png)

fixed

I074914 commented:
Behaviour of switch toggle is not working as expected. When we click on the "Default Semantic Compact Switch" and "Default Semantic Compact Switch" on first mouse click it go highlighted. and second click it will work. 
Expected: On first mouse click toggle should work.

I074914 commented:
> > Behaviour of switch toggle is not working as expected. When we click on the "Default Semantic Compact Switch" and "Default Semantic Compact Switch" on first click it is highlighted and second click it will work. But the issue is on first click switch toggle is not working.
> > Expected: On first mouse click toggle should work.
> 
> Fixed

Tested it is working fine. Thanks

manjunathanagaraj commented:
Looks fine to me. Accessbility testing will be verified by Purushotham.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/fundamental-ngx?pullRequest=2918) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/fundamental-ngx?pullRequest=2918) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/fundamental-ngx?pullRequest=2918) it.</sub>
